### PR TITLE
Update plugin system by using the PluginBase from D8

### PR DESCRIPTION
**Pull request by drupol**
Remove the tests related to the old BasePlugin.
Replaced the method getType() by getPluginId().

### CODE REVIEW

**Comment by drupol** [comment]
PHP Fatal error:  Class 'Drupal\Component\Plugin\PluginBase' not found in /home/travis/build/LionsAd/render_cache/src/RenderCache/Controller/AbstractBaseController.php on line 16

But the module service_container is a dependency.

**Comment by fabianx-ai** [comment]
Yeah, we first need to update service_container-7.x-1.x-dev on d.org, before this will work.

**Comment by fabianx-ai** [comment]
Will do so today, the PR looks great!

**Comment by drupol** [comment]
But I don't understand why the tests are failing. It should work if service_container (dev or not, because PluginBase is in there) is a dependency. What is wrong ?

**Comment by fabianx-ai** [comment]
service_container still calls the plugins with different arguments.

**Comment by fabianx-ai** [comment]
Oh and for unit tests we need the same fix of having tests/composer.json depend on service_container: 'dev-7.x-1.x'.

**Comment by fabianx-ai** [comment]
See tests/composer.json in OpenLayers.

**Comment by drupol** [comment]
Well done :+1: seems that tests are ok now... let's wait the end.

File: tests/composer.json [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+    "lionsad/service_container": "dev-7.x-1.x"
   },
   "autoload": {
     "psr-0": {
       "Drupal\\Core\\": "../lib/",
       "RenderCache": "../lib/",
-      "Drupal\\Component\\": "lib/"
+      "Drupal\\Component\\": "lib/",
+      "Drupal": "../lib/",
+      "Psr": "../lib/"
```
Comment (fabianx-ai) [inline tests/composer.json]: Can we revert those changes, please?

Comment (fabianx-ai) [inline tests/composer.json]: I mean just those 3 lines

**Comment by drupol** [comment]
Just updated code to have it working with latest version of Service Container, hopefully the syntax won't change from now.

**Comment by drupol** [comment]
1 timeout, 7 greens on 8 tests.

**Comment by fabianx-ai** [comment]
Thanks, @drupol

**Pull request merged**